pkg/clients/secretsmanager/secret.go

```diff
@@ -0,0 +1,267 @@
+		Namespace: s.Spec.ForProvider.SecretBinarySecretRef.Namespace,
+	}
+	sc := &corev1.Secret{}
+	if err := kube.Get(ctx, nn, sc); err != nil {
+		return nil, errors.Wrap(err, errGetSecretFailed)
+	}
+	if len(sc.Data) == 0 {
+		return nil, nil
+	}
+	scb, err := json.Marshal(sc.Data)
+	if err != nil {
+		return nil, errors.New(errSecretMarshalFailed)
+	}
+	return scb, nil
+}
+
+// GenerateCreateSecretsmanagerInput from ClusterParameters.
+func GenerateCreateSecretsmanagerInput(name string, p *v1alpha1.SecretParameters, secret []byte) *secretsmanager.CreateSecretInput {
+	c := &secretsmanager.CreateSecretInput{
+		Name:         awsclients.String(name),
+		SecretBinary: secret,
+	}
+
+	if p.Description != nil {
+		c.Description = p.Description
+	}
+
+	if p.KmsKeyID != nil {
+		c.KmsKeyId = p.KmsKeyID
+	}
+
+	if len(p.Tags) != 0 {
+		c.Tags = make([]secretsmanager.Tag, len(p.Tags))
+		for i, val := range p.Tags {
+			c.Tags[i] = secretsmanager.Tag{
+				Key:   aws.String(val.Key),
+				Value: aws.String(val.Value),
+			}
+		}
+	}
+
+	return c
+}
+
+// GenerateUpdateSecretInput from SecretParameters
+func GenerateUpdateSecretInput(name string, p v1alpha1.SecretParameters, secret []byte) *secretsmanager.UpdateSecretInput {
+	u := &secretsmanager.UpdateSecretInput{
+		SecretBinary: secret,
+		SecretId:     awsclients.String(name),
+	}
+
+	if p.Description != nil {
+		u.Description = p.Description
+	}
+
+	if p.KmsKeyID != nil {
+		u.KmsKeyId = p.KmsKeyID
+	}
+
+	return u
+}
+
+// GenerateDeleteSecretInput from SecretParameters
+func GenerateDeleteSecretInput(name string, p v1alpha1.SecretParameters) *secretsmanager.DeleteSecretInput {
+	d := &secretsmanager.DeleteSecretInput{
+		SecretId: awsclients.String(name),
+	}
+
+	switch r := GetRecoveryWindowInDays(p.RecoveryWindowInDays); r {
+	case 0:
+		d.ForceDeleteWithoutRecovery = boolToPtr(true)
+	default:
+		d.RecoveryWindowInDays = IntToPtr(int64(r))
+	}
+
+	return d
+}
+
+// UpdateObservation updates status.AtProvider
+func UpdateObservation(o *v1alpha1.SecretObservation, svr *secretsmanager.GetSecretValueResponse, do *secretsmanager.DescribeSecretOutput) {
+	if svr != nil {
+		svo := svr.GetSecretValueOutput
+		if svo != nil {
+			arn := svo.ARN
+			if svo.ARN == nil {
+				arn = do.ARN
+			}
+			o.ARN = awsclients.StringValue(arn)
+			o.VersionID = awsclients.StringValue(svo.VersionId)
+			o.VersionStages = svo.VersionStages
+
+			if svo.CreatedDate != nil {
+				o.CreatedDate = &metav1.Time{Time: *svo.CreatedDate}
+			}
+		}
+	}
+
+	if do != nil && do.DeletedDate != nil {
+		o.DeletedDate = &metav1.Time{Time: *do.DeletedDate}
+	}
+}
+
+// LateInitialize fills the empty fields in *v1alpha1.SecretParameters with the
+// values seen in secretsmanager.DescribeSecretOutput.
+func LateInitialize(in *v1alpha1.SecretParameters, so *secretsmanager.DescribeSecretOutput) { // nolint:gocyclo
+	if so == nil {
+		return
+	}
+	if in.Description == nil && so.Description != nil {
+		in.Description = so.Description
+	}
```

Comment:
```suggestion
	in.Description = awsclients.LateInitializeStringPtr(in.Description, so.Description)
```

You can use late-init helper functions if needed.